### PR TITLE
Make unsafe Wrenching of Pipes less... bonkers...

### LINKED ISSUE #12575: unwrenching a high presure air pipe sends you flying at absouloutly rediculos speeds
from the doors to the central hall loop on box to the arrivals vending row instantaneously.

### PR DESCRIPTION
**Pull request by neersighted**
Fixes #12575

:cl: neersighted
fix: Unwrenching atmos pipes is less... nuts...
fix: You don't get hopelessly spun when standing on a unwrenched pipe... Don't try it, though.
/:cl:

### CODE REVIEW

File: code/ATMOSPHERICS/atmospherics.dm [hunk 1]
```diff
@@ -147,31 +147,31 @@ Pipelines + Other Objects -> Pipe network
...
 	if(!user)
 		return
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/dir = get_dir(src, user)
```
Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: I liked `fuck_you_dir` tbh

File: code/ATMOSPHERICS/atmospherics.dm [hunk 2]
```diff
@@ -147,31 +147,31 @@ Pipelines + Other Objects -> Pipe network
...
+	if(!dir)
+		dir = rand(1,4)
+	var/turf/target = get_edge_target_turf(user, dir)
 
-	var/fuck_you_dir = get_dir(src,user)
-	var/turf/general_direction = get_edge_target_turf(user,fuck_you_dir)
 	user.visible_message("<span class='danger'>[user] is sent flying by pressure!</span>","<span class='userdanger'>The pressure sends you flying!</span>")
-	//Values based on 2*ONE_ATMOS (the unsafe pressure), resulting in 20 range and 4 speed
-	user.throw_at(general_direction,pressures/10,pressures/50)
+	user.throw_at(dir,  pressures/200, pressures/100)
```
Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: Leave speed alone; it's fine. Just change the range.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 3]
```diff
@@ -147,31 +147,31 @@ Pipelines + Other Objects -> Pipe network
...
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/dir = get_dir(src, user)
+	if(!dir)
+		dir = rand(1,4)
```
Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: This won't work, the cardinals are 1, 2, 4, and 8. Change this to `pick(cardinals) `.

Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: Alternatively, just revert this change. I liked the idea that you got thrown into the """ceiling""" if you unwrenched a pipe you were standing on.

File: code/ATMOSPHERICS/atmospherics.dm (hunk 1 above)
Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: `fuck_you_dunc`

File: code/ATMOSPHERICS/atmospherics.dm (hunk 3 above)
Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: I want people to get hurt, though...

File: code/ATMOSPHERICS/atmospherics.dm [hunk 4]
```diff
@@ -147,31 +147,33 @@ Pipelines + Other Objects -> Pipe network
...
+	var/turf/target = get_edge_target_turf(user, dir)
+	var/range = pressures/250
+	var/speed = range/5
 
-	var/fuck_you_dir = get_dir(src,user)
-	var/turf/general_direction = get_edge_target_turf(user,fuck_you_dir)
 	user.visible_message("<span class='danger'>[user] is sent flying by pressure!</span>","<span class='userdanger'>The pressure sends you flying!</span>")
-	//Values based on 2*ONE_ATMOS (the unsafe pressure), resulting in 20 range and 4 speed
-	user.throw_at(general_direction,pressures/10,pressures/50)
+	user.throw_at(target, range, speed)
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: you can't ```throw_at()``` a dir, it has to be an object, hence the furthest tile on this Z in the direction of ```fuck_you_dir```

File: code/ATMOSPHERICS/atmospherics.dm (hunk 1 above)
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: put it back, ```fuck_you_dir``` is a fan favourite var.
that *I* made!

File: code/ATMOSPHERICS/atmospherics.dm (hunk 4 above)
Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: Yeah, just fixed the typo.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 5]
```diff
@@ -147,31 +147,33 @@ Pipelines + Other Objects -> Pipe network
...
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/dir = get_dir(src, user)
+	if(!dir)
+		dir = pick(cardinals)
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: cardinal, not cardinals

File: code/ATMOSPHERICS/atmospherics.dm (hunk 1 above)
Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: `fuck_you_remie`

Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: I will CLOSE THIS PR young man! do as you're told or I'll take away your NanoUI and your video games!

Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: Mommy why ;-;

Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: funny code/comments where they are functional but still funny are one of our greatest joys, we reminisce ALL the time about "hey remember there's this comment in this file" "Haha yeah omg"

File: code/ATMOSPHERICS/atmospherics.dm (hunk 5 above)
Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: My bad

File: code/ATMOSPHERICS/atmospherics.dm (hunk 1 above)
Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: Okay fine

File: code/ATMOSPHERICS/atmospherics.dm [hunk 6]
```diff
@@ -147,31 +147,33 @@ Pipelines + Other Objects -> Pipe network
...
 	if(!user)
 		return
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/fuck_you_remie = get_dir(src, user)
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/atmospherics.dm]: >:/

Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: Non descriptive and unhelpful var name
DENIED

File: code/ATMOSPHERICS/atmospherics.dm [hunk 7]
```diff
@@ -147,31 +147,33 @@ Pipelines + Other Objects -> Pipe network
...
 	if(!user)
 		return
-
 	if(!pressures)
 		var/datum/gas_mixture/int_air = return_air()
 		var/datum/gas_mixture/env_air = loc.return_air()
-		pressures = int_air.return_pressure()-env_air.return_pressure()
+		pressures = int_air.return_pressure() - env_air.return_pressure()
+
+	var/fuck_you_dir = get_dir(src, user) // God damnit, Remie.
```
Comment (JJRcop) [inline code/ATMOSPHERICS/atmospherics.dm]: hooo boy when oranges gets a look at this comment

Comment (optimumtact) [inline code/ATMOSPHERICS/atmospherics.dm]: yes please remove this useless comment

Comment (neersighted) [inline code/ATMOSPHERICS/atmospherics.dm]: ...

Comment (optimumtact) [inline code/ATMOSPHERICS/atmospherics.dm]: TEAR IT OUT, RIP AND TEAR IT'S GUTS, OR I'LL TEAR IT OUT FOR YOU

Comment (duncathan) [inline code/ATMOSPHERICS/atmospherics.dm]: Comments are for code comprehension

Comment (Iamgoofball) [inline code/ATMOSPHERICS/atmospherics.dm]: comments are for comprehending the behemoth that is your mother

**Comment by Cheridan** [comment]
Work on your commit logs

**Comment by neersighted** [comment]
...

**Comment by duncathan** [comment]
Reminder that at least two of those would've been avoided if you hadn't been difficult @neersighted
